diff -ruN old/build-aux/config.sub new/build-aux/config.sub
--- old/build-aux/config.sub	2022-10-09 17:42:28.000000000 +0300
+++ new/build-aux/config.sub	2024-01-17 13:26:57.745053162 +0300
@@ -1758,7 +1758,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* )
+	     | fiwix* | mlibc* | argentum* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
