diff -ruN orig/bfd/config.bfd new/bfd/config.bfd
--- orig/bfd/config.bfd	2023-07-03 02:00:00.000000000 +0300
+++ new/bfd/config.bfd	2024-01-01 10:23:03.587004723 +0300
@@ -411,6 +411,10 @@
     targ_defvec=arm_elf32_le_vec
     targ_selvecs=arm_elf32_be_vec
     ;;
+  arm-*-osdev*)
+    targ_defvec=arm_elf32_le_vec
+    targ_selvecs=arm_elf32_be_vec
+    ;;
   armeb-*-elf | arm*b-*-freebsd* | arm*b-*-linux-* | armeb-*-eabi*)
     targ_defvec=arm_elf32_be_vec
     targ_selvecs=arm_elf32_le_vec
diff -ruN orig/config.sub new/config.sub
--- orig/config.sub	2023-07-03 02:00:00.000000000 +0300
+++ new/config.sub	2024-01-01 10:21:15.717006706 +0300
@@ -1758,7 +1758,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* )
+	     | fiwix* | mlibc* | osdev* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ruN orig/gas/configure.tgt new/gas/configure.tgt
--- orig/gas/configure.tgt	2023-07-03 02:00:00.000000000 +0300
+++ new/gas/configure.tgt	2024-01-01 10:25:47.027004698 +0300
@@ -167,6 +167,7 @@
   arm-*-pe)				fmt=coff em=pe ;;
   arm-*-fuchsia*)			fmt=elf ;;
   arm-*-haiku*)				fmt=elf em=haiku ;;
+  arm-*-osdev*)			fmt=elf ;;
 
   avr-*-*)				fmt=elf bfd_gas=yes ;;
 
diff -ruN orig/ld/Makefile.am new/ld/Makefile.am
--- orig/ld/Makefile.am	2023-07-03 02:00:00.000000000 +0300
+++ new/ld/Makefile.am	2024-01-01 11:05:34.662497346 +0300
@@ -173,6 +173,7 @@
 	earmelf_linux_fdpiceabi.c \
 	earmelf_nacl.c \
 	earmelf_nbsd.c \
+	earmelf_osdev.c \
 	earmelf_phoenix.c \
 	earmelf_vxworks.c \
 	earmelfb.c \
@@ -671,6 +672,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_linux_fdpiceabi.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nacl.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nbsd.Pc@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_osdev.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_phoenix.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_vxworks.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelfb.Pc@am__quote@
diff -ruN orig/ld/Makefile.in new/ld/Makefile.in
--- orig/ld/Makefile.in	2023-07-03 02:00:00.000000000 +0300
+++ new/ld/Makefile.in	2024-01-01 11:12:54.002497746 +0300
@@ -674,6 +674,7 @@
 	earmelf_linux_fdpiceabi.c \
 	earmelf_nacl.c \
 	earmelf_nbsd.c \
+	earmelf_osdev.c \
 	earmelf_phoenix.c \
 	earmelf_vxworks.c \
 	earmelfb.c \
@@ -1293,6 +1294,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_linux_fdpiceabi.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nacl.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nbsd.Po@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_osdev.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_phoenix.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_vxworks.Po@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelfb.Po@am__quote@
@@ -2388,6 +2390,7 @@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_linux_fdpiceabi.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nacl.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_nbsd.Pc@am__quote@
+@AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_osdev.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_phoenix.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelf_vxworks.Pc@am__quote@
 @AMDEP_TRUE@@am__include@ @am__quote@./$(DEPDIR)/earmelfb.Pc@am__quote@
diff -ruN orig/ld/configure.tgt new/ld/configure.tgt
--- orig/ld/configure.tgt	2023-07-03 02:00:00.000000000 +0300
+++ new/ld/configure.tgt	2024-01-01 10:57:31.262497332 +0300
@@ -194,6 +194,8 @@
 			;;
 arm-*-phoenix*)		targ_emul=armelf
 			;;
+arm-*-osdev*)		targ_emul=armelf_osdev
+			;;
 armeb-*-elf | armeb-*-eabi*)
 			targ_emul=armelfb
 			;;
diff -ruN orig/ld/emulparams/armelf_osdev.sh new/ld/emulparams/armelf_osdev.sh
--- orig/ld/emulparams/armelf_osdev.sh	1970-01-01 03:00:00.000000000 +0300
+++ new/ld/emulparams/armelf_osdev.sh	2024-01-01 12:11:07.972498196 +0300
@@ -0,0 +1,2 @@
+source_sh ${srcdir}/emulparams/armelf.sh
+DATA_ADDR="ALIGN(${MAXPAGESIZE})"
