diff -ruN old/build-aux/config.sub new/build-aux/config.sub
--- old/build-aux/config.sub	2024-04-20 14:11:52.000000000 +0300
+++ new/build-aux/config.sub	2024-08-17 17:33:07.356971319 +0300
@@ -1768,7 +1768,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* | ironclad* )
+	     | fiwix* | mlibc* | cos* | mbr* | ironclad* | argentum* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
