diff -ruN old/config.sub new/config.sub
--- old/config.sub	2020-12-31 17:34:29.000000000 +0300
+++ new/config.sub	2024-01-19 15:37:57.306271164 +0300
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -osdev*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
