diff -ruN old/build-aux/config.sub new/build-aux/config.sub
--- old/build-aux/config.sub	2023-08-11 16:01:01.000000000 +0300
+++ new/build-aux/config.sub	2024-06-29 16:58:02.324643650 +0300
@@ -1736,7 +1736,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* | mlibc* | cos* | mbr* )
+	     | fiwix* | mlibc* | cos* | mbr* | argentum* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ruN old/configure new/configure
--- old/configure	2023-08-29 16:37:00.000000000 +0300
+++ new/configure	2024-06-29 16:59:26.824629322 +0300
@@ -63291,6 +63291,9 @@
             # Guess no on native Windows.
           mingw* | windows*)
             gl_cv_func_nanosleep='guessing no' ;;
+            # Guess yes on Argentum
+          argentum*)
+            gl_cv_func_nanosleep='guessing yes' ;;
             # If we don't know, obey --enable-cross-guesses.
           *)
             gl_cv_func_nanosleep="$gl_cross_guess_normal" ;;
diff -ruN old/src/ls.c new/src/ls.c
--- old/src/ls.c	2023-08-29 14:39:27.000000000 +0300
+++ new/src/ls.c	2024-06-29 16:58:02.324643650 +0300
@@ -4268,7 +4268,12 @@
       return MAX (0, len);
     }
   else
-    return snprintf (nullptr, 0, "%"PRIuMAX, id);
+    {
+      char *ptr;
+      size_t n;
+      free(asnprintf(NULL, &n, "%"PRIuMAX, id));
+      return n;
+    }
 }
 
 /* Return the number of columns that format_user will print.  */
