diff -ruN orig/build-aux/config.sub new/build-aux/config.sub
--- orig/build-aux/config.sub	2022-04-07 05:15:05.000000000 +0300
+++ new/build-aux/config.sub	2023-05-20 11:25:35.793296907 +0300
@@ -1754,7 +1754,7 @@
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
 	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | wasi* \
 	     | nsk* | powerunix* | genode* | zvmoe* | qnx* | emx* | zephyr* \
-	     | fiwix* )
+	     | fiwix* | osdev* )
 		;;
 	# This one is extra strict with allowed versions
 	sco3.2v2 | sco3.2v[4-9]* | sco5v6*)
diff -ruN orig/lib/fpurge.c new/lib/fpurge.c
--- orig/lib/fpurge.c	2022-01-01 22:04:48.000000000 +0300
+++ new/lib/fpurge.c	2023-05-20 15:43:47.026223635 +0300
@@ -143,7 +143,13 @@
   fp->rp = fp->wp = fp->lp = fp->buf;
   return 0;
 # else
-#  error "Please port gnulib fpurge.c to your platform! Look at the definitions of fflush, setvbuf and ungetc on your system, then report this to bug-gnulib."
+  if (fp->read_end != NULL)
+    fp->next = fp->read_end;
+  else if (fp->write_end != NULL)
+    fp->next = fp->write_end;
+  fp->back_count = 0;
+  fp->read_save = NULL;
+  return 0;
 # endif
 
 #endif
diff -ruN orig/lib/freading.c new/lib/freading.c
--- orig/lib/freading.c	2022-01-01 22:04:48.000000000 +0300
+++ new/lib/freading.c	2023-05-20 15:43:34.972890620 +0300
@@ -69,7 +69,7 @@
     return 0;
   return (fp->state == 3 /* RD */ && (fp->bufl == 0 || fp->rp < fp->wp));
 # else
-#  error "Please port gnulib freading.c to your platform!"
+  return ((fp->mode & _MODE_READ) && (fp->write_end == NULL));
 # endif
 }
 
diff -ruN orig/lib/fseeko.c new/lib/fseeko.c
--- orig/lib/fseeko.c	2022-01-01 22:04:48.000000000 +0300
+++ new/lib/fseeko.c	2023-05-20 15:43:22.619557615 +0300
@@ -108,7 +108,7 @@
      bug-gnulib.  */
   if (0)
 #else
-  #error "Please port gnulib fseeko.c to your platform! Look at the code in fseeko.c, then report this to bug-gnulib."
+  if (0)
 #endif
     {
       /* We get here when an fflush() call immediately preceded this one (or
diff -ruN orig/lib/fseterr.c new/lib/fseterr.c
--- orig/lib/fseterr.c	2022-01-01 22:04:48.000000000 +0300
+++ new/lib/fseterr.c	2023-05-20 15:43:05.886224727 +0300
@@ -79,6 +79,6 @@
     }
   errno = saved_errno;
 #else
- #error "Please port gnulib fseterr.c to your platform! Look at the definitions of ferror and clearerr on your system, then report this to bug-gnulib."
+  fp->state |= _STATE_ERROR;
 #endif
 }
